docs/Errors.md

```diff
@@ -37,7 +37,16 @@ To customize this behavior you should use [`setErrorHandler`](Server.md#seterror
 From the [Hooks documentation](Hooks.md#manage-errors-from-a-hook): 
 > If you get an error during the execution of your hook, just pass it to `done()` and Fastify will automatically close the request and send the appropriate error code to the user.
 
-If you have defined a custom error handler using `setErrorHandler` the error will be routed there, otherwise it will be routed to Fastify’s generic error handler. Fastify's generic error handler will use the header and status code in the Error object if it exists. The headers and status code will not be automatically set if a custom error handler is provided.
+If you have defined a custom error handler using `setErrorHandler` the error will
+be routed there. If `setErrorHandler` has been called multiple times, the error will be
+routed to the precedent error handler if the error handler or an hook `throw` or
+call `reply.send(error)`.
+Error handlers are fully encapsulated, so a `setErrorHandler` call in an encapsulated
+plugin will limit its effect there.
```

Comment:
Do we speak about `plugin` or `context` here? The encapsulation documentation speaks about `context`. Might be confusing

Reply:
How would you rephrase? The current text sound clear to me but I'm definitely not the target audience for this text.

Reply:
I don't have a better phrasing to be honest. @jsumners might have an idea ?

Reply:
See my suggestion below.